libcalico-go/lib/backend/k8s/resources/ipam_affinity.go

```diff
@@ -210,6 +255,44 @@ func (c *blockAffinityClient) DeleteKVP(ctx context.Context, kvp *model.KVPair)
 	return c.toV1(kvp)
 }
 
+func (c *blockAffinityClient) deleteKVPV3(ctx context.Context, kvp *model.KVPair) (*model.KVPair, error) {
+	// We need to mark as deleted first, since the Kubernetes API doesn't support
+	// compare-and-delete. This update operation allows us to eliminate races with other clients.
+	var err error
+	log.Debugf("This is the value: %+v: bool: %t", kvp.Key, kvp.Value == nil)
```

Comment:
can remove this log now :grin:

Reply:
Actually, `deleteKVPV3` can really just be a "NotImplemented" error, right? We don't support delete on the v3 API. And given we don't have the Value in hand... I don't think we ever will :laughing: :sob:

Reply:
I think we need to have an implementation here since using this backs the `clientv3` for the Kubernetes datastore. I can get rid of it if we do not plan on eventually standardizing IPAM code on the `clientv3` library, but if we do, then we will probably need this.